test/cache_invalidation_test.rb

```diff
@@ -211,4 +211,166 @@ def test_cache_not_expired_until_after_transaction
     assert_match(/^COMMIT\b/, sql_queries[2])
     assert_match(/^cache_write.active_support /, log.last.last)
   end
+
+  def test_expire_cache_with_primary_key_only
+    assert_equal(@record, Item.fetch(1))
+    refute_equal(IdentityCache::DELETED, read_entity)
+    assert(@record.expire_cache)
+    assert_equal(IdentityCache::DELETED, read_entity)
+  end
+
+  def test_expire_cache_with_primary_key_only_on_failure
+    assert_equal(@record, Item.fetch(1))
+    refute_equal(IdentityCache::DELETED, read_entity)
+
+    with_cache_backend do
```

Comment:
This does the opposite of what it looks like from the callsite. It would be more readable to explicitly use the CacheConnection.unconnected_cache_backend argument